src/SymmetricPositiveDefinite.jl

```diff
@@ -0,0 +1,272 @@
+"""
+log!(P::SymmetricPositiveDefinite{N}, v, x, y) where N = log!(MetricManifold(P,LinearAffineMetric()),y,x,v)
+@doc doc"""
+    log!(P,v,x,y)
+
+compute the exponential map from `x` to `y` on the [`SymmetricPositiveDefinite`](@ref)
+as a [`MetricManifold`](@ref) with [`LinearAffineMetric`](@ref) and modify `v`. The formula reads
+
+```math
+\log_x y = x^{\frac{1}{2}}\operatorname{Log}(x^{-\frac{1}{2}} y x^{-\frac{1}{2}})x^{\frac{1}{2}},
+```
+where $\operatorname{Log}$ denotes to the matrix logarithm.
+"""
+function log!(P::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, v, x, y) where N
+    e = eigen(Symmetric(x))
+    U = e.vectors
+    S = e.values
+    Ssqrt = Diagonal( sqrt.(S) )
+    SsqrtInv = Diagonal( 1 ./ sqrt.(S) )
+    xSqrt = U*Ssqrt*transpose(U)
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    T = xSqrtInv * y * xSqrtInv
+    e2 = eigen( (T + transpose(T))/2 )
+    Se = Diagonal( log.(max.(e2.values,eps()) ) )
+    Ue = e2.vectors
+    v = xSqrt * Ue*Se*transpose(Ue) * xSqrt
+    return v
+end
+
+function representation_size(::SymmetricPositiveDefinite{N}, ::Type{T}) where {N, T<:Union{MPoint, TVector, CoTVector}}
+    return (N,N)
+end
+
+
+@doc doc"""
+    vector_transport(P,vto,x,v,y,::ParallelTransport)
+
+compute the parallel transport on the [`SymmetricPositiveDefinite`](@ref) with its default metric, [`LinearAffineMetric`](@ref).
+"""
+vector_transport!(P::SymmetricPositiveDefinite{N},vto, x, v, y, m) where N = vector_transport!(MetricManifold(P,LinearAffineMetric()),vto, x, v, y, m)
+@doc doc"""
+    vector_transport!(P,vto,x,v,y,::ParallelTransport)
+
+compute the parallel transport on the [`SymmetricPositiveDefinite`](@ref) as a [`MetricManifold`](@ref) with the [`LinearAffineMetric`](@ref).
+The formula reads
+
+```math
+P_{x\to y}(v) = x^{\frac{1}{2}}
+\operatorname{Exp}\bigl(
+\frac{1}{2}x^{-\frac{1}{2}}\log_x(y)x^{-\frac{1}{2}}
+\bigr)
+x^{-\frac{1}{2}}v x^{-\frac{1}{2}}
+\operatorname{Exp}\bigl(
+\frac{1}{2}x^{-\frac{1}{2}}\log_x(y)x^{-\frac{1}{2}}
+\bigr)
+x^{\frac{1}{2}},
+```
+
+where $\operatorname{Exp}$ denotes the matrix exponential
+and `log` the logarithmic map.
+"""
+function vector_transport!(::MetricManifold{SymmetricPositiveDefinite{N},LinearAffineMetric}, vto, x, v, y, ::ParallelTransport) where N
+    if norm(x-y)<1e-13
+        vto = v
+        return vto
+    end
+    e = eigen(Symmetric(x))
+    U = e.vectors
+    S = e.values
+    Ssqrt = sqrt.(S)
+    SsqrtInv = Diagonal( 1 ./ Ssqrt )
+    Ssqrt = Diagonal( Ssqrt )
+    xSqrt = U*Ssqrt*transpose(U)
+    xSqrtInv = U*SsqrtInv*transpose(U)
+    tv = xSqrtInv * v * xSqrtInv
+    ty = xSqrtInv * y * xSqrtInv
+    e2 = eigen( ty )
+    Se = Diagonal( log.(e2.values) )
+    Ue = e2.vectors
+    ty2 = Ue*Se*transpose(Ue)
+    e3 = eigen( ty2 )
+    Sf = Diagonal( exp.(e3.values) )
+    Uf = e3.vectors
+    vto = xSqrt*Uf*Sf*transpose(Uf)*(0.5*(tv+transpose(tv)))*Uf*Sf*transpose(Uf)*xSqrt
```

Comment:
And this needs broadcasting too.

Reply:
Again, I'd wrap everything that is known to by symmetric by `Symmetric`: `xSqrt`, `xSqrtInv `, `tv`, `ty`, `ty2`, and cache `xSqrt*Uf*Sf*transpose(Uf)` as above, and then use the `copyto!`.